### PR TITLE
feat(provider/google): Add support for Shared VPC Networking (XPN).

### PR DESCRIPTION
**Pull request by duftler**

### CODE REVIEW

**Comment by duftler** [comment]
Relates to: https://github.com/spinnaker/clouddriver/pull/1691

**Comment by duftler** [comment]
The various locations throughout the ui that render a network/subnet as either a pulldown choice, or as an attribute in a details view, will all now decorate the network/subnet resource with the XPN hostname when appropriate.

For example:
image

**Comment by duftler** [comment]
image

File: app/scripts/modules/google/src/serverGroup/configure/wizard/securityGroups/tagManager.service.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ module.exports = angular.module('spinnaker.deck.gce.tagManager.service', [])
     this.inferSecurityGroupIdsFromTags = (commandTags = []) => {
       return _.chain(commandTags)
         .map(t => this.securityGroupObjectsKeyedByTag[t.value])
+        .filter(sg => sg.network === this.command.network)
```
Comment (duftler) [inline app/scripts/modules/google/src/serverGroup/configure/wizard/securityGroups/tagManager.service.js]: All the tag resolution logic needed to filter by network as well.

File: app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts [hunk 2]
```diff
@@ -250,6 +252,7 @@ module(GCE_INTERNAL_LOAD_BALANCER_CTRL, [
     GCE_COMMON_LOAD_BALANCER_COMMAND_BUILDER,
     ACCOUNT_SERVICE,
     INFRASTRUCTURE_CACHE_SERVICE,
+    require('../../../common/xpnNaming.gce.service.js'),
```
Comment (danielpeach) [inline app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts]: `require('google/common/xpnNaming.gce.service.js')` would be a little easier to read.

File: app/scripts/modules/google/src/securityGroup/details/securityGroupDetail.html [hunk 3]
```diff
@@ -48,6 +48,16 @@ <h3 select-on-dbl-click>
           </render-if-feature>
         </ul>
       </div>
+      <div ng-if="securityGroup.id.includes('/')" class="dropdown" uib-dropdown dropdown-append-to-body>
+        <button type="button" class="btn btn-sm btn-primary dropdown-toggle" uib-dropdown-toggle>
+          Security Group Actions <span class="caret"></span>
+        </button>
+        <ul uib-tooltip="You cannot modify shared VPC host project firewall rules." class="dropdown-menu" uib-dropdown-menu role="menu">
+          <li class="disabled"><a href ng-click="ctrl.editInboundRules()">Edit Inbound Rules</a></li>
```
Comment (danielpeach) [inline app/scripts/modules/google/src/securityGroup/details/securityGroupDetail.html]: Might as well remove the `ng-click` attributes.

File: app/scripts/modules/google/src/common/xpnNaming.gce.service.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+import _ from 'lodash';
+
+const angular = require('angular');
+
+module.exports = angular.module('spinnaker.common.xpnNaming.gce.service', [])
```
Comment (danielpeach) [inline app/scripts/modules/google/src/common/xpnNaming.gce.service.js]: Module name should be `spinnaker.gce.common.xpnNaming.service`.

File: app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts (hunk 2 above)
Comment (duftler) [inline app/scripts/modules/google/src/loadBalancer/configure/internal/gceCreateInternalLoadBalancer.controller.ts]: Done.

File: app/scripts/modules/google/src/securityGroup/details/securityGroupDetail.html (hunk 3 above)
Comment (duftler) [inline app/scripts/modules/google/src/securityGroup/details/securityGroupDetail.html]: Done.

File: app/scripts/modules/google/src/common/xpnNaming.gce.service.js (hunk 4 above)
Comment (duftler) [inline app/scripts/modules/google/src/common/xpnNaming.gce.service.js]: Done.

**Pull request merged**